src/global/mod.rs

```diff
@@ -22,33 +26,30 @@ fn find_executables(prefix: &Prefix, prefix_package: &PrefixRecord) -> Vec<PathB
         .collect()
 }
 
-fn is_executable(prefix: &Prefix, relative_path: &Path) -> bool {
-    // Check if the file is in a known executable directory.
-    let binary_folders = if cfg!(windows) {
-        &([
-            "",
-            "Library/mingw-w64/bin/",
-            "Library/usr/bin/",
-            "Library/bin/",
-            "Scripts/",
-            "bin/",
-        ][..])
-    } else {
-        &(["bin"][..])
-    };
-
-    let parent_folder = match relative_path.parent() {
-        Some(dir) => dir,
-        None => return false,
-    };
-
-    if !binary_folders
+/// Processes prefix records (that you can get by using `find_installed_packages`)
+/// to filter and collect executable files.
+pub fn find_executables_for_many_records(
+    prefix: &Prefix,
+    prefix_packages: &[PrefixRecord],
+) -> Vec<Executable> {
+    let executables = prefix_packages
         .iter()
-        .any(|bin_path| Path::new(bin_path) == parent_folder)
-    {
-        return false;
-    }
+        .flat_map(|record| {
+            record
+                .files
+                .iter()
+                .filter(|relative_path| is_executable(prefix, relative_path))
+                .filter_map(|path| {
+                    path.iter().last().and_then(OsStr::to_str).map(|name| {
+                        Executable::new(strip_executable_extension(name.to_string()), path.clone())
+                    })
```

Comment:
We have `pixi_utils::executable_from_path` as a function to do the conversion from path the an executable string.